File: mock-facility-management/src/main/java/de/vkb/dojo/es/facilityManagement/model/feedback/SuccessFeedback.java
package de.vkb.dojo.es.facilityManagement.model.feedback;

import com.fasterxml.jackson.annotation.*;
import de.vkb.dojo.es.facilityManagement.model.feedback.ref.Reference;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("success")
public class SuccessFeedback implements Feedback {
    private final Reference reference;

    @JsonCreator
    public SuccessFeedback(
            @JsonProperty("reference") Reference reference
    ) {
        this.reference = reference;
    }

    @JsonGetter
    public Reference getReference() {
        return reference;
    }

    public Boolean getSuccess() {
        return true;
    }
}

File: mock-human-resources/src/main/kotlin/de/vkb/dojo/es/humanResources/model/feedback/SuccessFeedback.kt
package de.vkb.dojo.es.humanResources.model.feedback

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonTypeName
import de.vkb.dojo.es.humanResources.model.ref.Reference

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("success")
data class SuccessFeedback(
    val reference: Reference
): Feedback {
    override val success = true
}

